module-1/lab-functional-programming/your-code/Q3.ipynb

```diff
@@ -13,7 +13,7 @@
   },
```

Comment:
```Python
#your code
import langdetect
from functools import reduce
words = ['good morning', '早上好', 'доброго', 'おはようございます', 'everyone', '大家', 'каждый', 'みんな']

engTerms=reduce((lambda x,y: x+""+y if langdetect.detect(x)=='en' else x), words)
#I don understand why the above does not work
print(engTerms)
# Just changing the variable for detect because you are not using y variable in any part of your lambda
engTerms=reduce((lambda x,y: x+""+y if langdetect.detect(y)=='en' else x), words)
#Other way of solving this question: 
import langdetect
from functools import reduce
words = ['good morning', '早上好', 'доброго', 'おはようございます', 'everyone', '大家', 'каждый', 'みんな']

reduce((lambda x, y: x if not langdetect.detect(y)=='en' else ''.join([x, ' ', y])), words)
```